.github/workflows/pytest-core-nompi.yml

```diff
@@ -121,21 +129,26 @@ jobs:
       with:
         python-version: ${{ matrix.python-version }}
 
-    - name: Build docker image
-      if: contains(matrix.name, 'docker')
+    - name: Build docker image for GCC variants
+      if: "contains(matrix.name, 'docker') && contains(matrix.name, 'gcc')"
       run: |
           docker build . --file docker/Dockerfile.devito --tag devito_img --build-arg base=devitocodes/bases:cpu-${{ matrix.arch }}
 
+    - name: Build docker image for INTEL variants
```

Comment:
Don't really like to duplicate that step makes a lot of "single case" steps which arent' really good

Reply:
>Maybe we can add the tags icc/icx to the dovcker build CI so we only build it once but add aliases for easy pull


For some reason I had not seen this command, indeed it solves the problems stated form you and Fabio